File: 30_days_C#_HackerRank_Challenge/LinkedList.cs
using System;
class Node
{
	public int data;
	public Node next;
    public Node(int d){
        data=d;
        next=null;
    }
		
}
class Solution {
    public static Node insert(Node head, int data) {
        Node newNode = new Node(data);

        // If the list is empty, the new node becomes the head
        if (head == null) {
            return newNode;
        }

        // Traverse to the end of the list
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }

        // Link the last node to our new node
        current.next = newNode;
        return head;
    }


	public static void display(Node head)
	{
		Node start=head;
		while(start!=null)
		{
			Console.Write(start.data+" ");
			start=start.next;
		}
	}
    static void Main(String[] args) {
	
		Node head=null;
        int T=Int32.Parse(Console.ReadLine());
        while(T-->0){
            int data=Int32.Parse(Console.ReadLine());
            head=insert(head,data);
        }
		display(head);
	}
}
	

File: 30_days_C++(modern)_HackerRank_Challenge/LinkedList.cpp
#CPP 11 14
#include <iostream>
#include <cstddef>
using namespace std;	
class Node
{
    public:
        int data;
        Node *next;
        Node(int d){
            data=d;
            next=NULL;
        }
};
class Solution{
    public:

      Node* insert(Node *head, int data) {
        // Use 'new' in C++ to call the constructor properly
        Node* newNode = new Node(data);

        // If the list is empty, the new node is the head
        if (head == NULL) {
            return newNode;
        }

        // Traverse to the end of the list
        Node* current = head;
        while (current->next != NULL) {
            current = current->next;
        }

        // Link the last node to the new node
        current->next = newNode;
        return head;
    }

      void display(Node *head)
      {
          Node *start=head;
          while(start)
          {
              cout<<start->data<<" ";
              start=start->next;
          }
      }
};
int main()
{
	Node* head=NULL;
  	Solution mylist;
    int T,data;
    cin>>T;
    while(T-->0){
        cin>>data;
        head=mylist.insert(head,data);
    }	
	mylist.display(head);
		
}

#CPP 20
#include <iostream>
#include <cstddef>
using namespace std;

class Node {
public:
    int data;
    Node *next;
    Node(int d) {
        data = d;
        next = NULL;
    }
};

class Solution {
public:
    Node* insert(Node *head, int data) {
        // Use 'new' in C++ to call the constructor properly
        Node* newNode = new Node(data);

        // If the list is empty, the new node is the head
        if (head == NULL) {
            return newNode;
        }

        // Traverse to the end of the list
        Node* current = head;
        while (current->next != NULL) {
            current = current->next;
        }

        // Link the last node to the new node
        current->next = newNode;
        return head;
    }

    void display(Node *head) {
        Node *start = head;
        while (start) {
            cout << start->data << " ";
            start = start->next;
        }
    }
}; // Closing the Solution class properly

int main() {
    Node* head = NULL;
    Solution mylist;
    int T, data;
    cin >> T;
    while (T-- > 0) {
        cin >> data;
        head = mylist.insert(head, data);
    }
    mylist.display(head);
    return 0;
}
